client-core/src/service/wallet_state_service.rs

```diff
@@ -192,20 +187,19 @@ impl Default for WalletState {
 
 impl WalletState {
     /// Applies memento to wallet state
-    fn apply_memento(&mut self, memento: &WalletStateMemento) -> Result<()> {
+    pub fn apply_memento(&mut self, memento: &WalletStateMemento) {
         for operation in memento.0.iter() {
-            self.apply_memento_operation(operation)?
+            self.apply_memento_operation(operation);
         }
-
-        Ok(())
     }
 
     /// Applies a memento operation to wallet state
-    fn apply_memento_operation(&mut self, memento_operation: &MementoOperation) -> Result<()> {
+    fn apply_memento_operation(&mut self, memento_operation: &MementoOperation) {
```

Comment:
This function should not be infallible.

Reply:
Other then the `Coin` addition overflow, there are no other error condition in this function.